src/components/app/UrlManager.js

```diff
@@ -144,13 +144,31 @@ class UrlManager extends React.PureComponent<Props> {
       }
     }
 
-    if (
-      previousUrlState.dataSource !== newUrlState.dataSource ||
-      previousUrlState.hash !== newUrlState.hash
-    ) {
-      // Profile sanitization and publishing can do weird things for the history API.
-      // Rather than write lots of complicated interactions, just prevent the back button
-      // from working when going between a published profile, and one that is not.
+    // Profile sanitization and publishing can do weird things for the history API
+    // and we could end up having unconsistent state.
+    // That's why we prevent going back in history in these cases:
+    // 1 - between "from-addon" and "public" (in any direction)
+    // 2 - with the "public" datasource when the hash changes (this means the
+    //     user once published again an already public profile, and then wants to go
+    //     back).
+    // But we want to accept the other interactions.
+
+    // 1. Do we move between "from-addon" and "public"?
```

Comment:
Thanks for the comments, this helps explains things.

No action needed, just thinking: I wonder if we could describe valid interactions in a more declarative way, with some kind of state machine description that is also type checked to make sure we are handling all the cases.